File: dastanlib/src/main/java/com/dastanapps/dastanlib/chat/base/ChatBase.java
package com.dastanapps.dastanlib.chat.base;

import com.dastanapps.dastanlib.DastanApp;
import com.dastanapps.dastanlib.chat.DChatXMPP;
import com.dastanapps.dastanlib.chat.service.XMPPService;

/**
 * Created by DastanIqbal on 7/10/16.
 */

public class ChatBase {
    protected XMPPService xmppService = XMPPService.getXMPPService();
    protected DChatXMPP dastanXMPP = XMPPService.getDastanXmppConfig();

    protected void ensureConnection() {
        if (xmppService != null) {
            xmppService.start();
        } else {
            XMPPService.runService(DastanApp.getInstance());
        }
    }
}

File: chat/src/main/java/com/dastanapps/dastanlib/chat/base/ChatBase.kt
package com.dastanapps.dastanlib.chat.base

import com.dastanapps.dastanlib.DastanLibApp
import com.dastanapps.dastanlib.chat.service.XMPPService

/**
 * Created by DastanIqbal on 7/10/16.
 */

open class ChatBase {
    protected var xmppService: XMPPService? = XMPPService.getXMPPService()
    protected var dastanXMPP = XMPPService.getDastanXmppConfig()

    protected fun ensureConnection() {
        if (xmppService != null) {
            xmppService?.start()
        } else {
            XMPPService.runService(DastanLibApp.INSTANCE)
        }
    }
}
